### PR TITLE
Poly payments for ticker and ST registration

### PR DESCRIPTION
**Pull request by thegostep**
- added poly payment requirement to use `registerTicker` and `generateSecurityToken`
- initial registration fee is set in contract constructor and can be changed via `setPolyRegistrationFee`
- `ReclaimFunds.sol` can be inherited by contracts to allow owner to withdraw funds
- added payable fallback function to each registry contract
- modified the deployment process for each test file to accommodate new features
- minor changes to logs of tests to facilitate readability

### CODE REVIEW

File: contracts/SecurityTokenRegistry.sol [hunk 1]
```diff
@@ -83,6 +95,14 @@ contract SecurityTokenRegistry is Ownable, ISecurityTokenRegistry, Util {
         protocolVersionST[_version] = _stVersionProxyAddress;
     }
 
+    /**
+     * @dev set the ticker registration fee in POLY tokens
+     * @param _registrationFee registration fee in POLY tokens (base 18 decimals)
+     */
+    function setPolyRegistrationFee(uint256 _registrationFee) public onlyOwner {
```
Comment (satyamakgec) [inline contracts/SecurityTokenRegistry.sol]: change the function name as we are changing the registration fee, not setting the fee for one time
it should be named as `changePolyRegistrationFee` and also add the event for logging the same. with old value and new value as the arguments of the event.

File: contracts/TickerRegistry.sol [hunk 2]
```diff
@@ -86,6 +100,14 @@ contract TickerRegistry is ITickerRegistry, Ownable, Util {
         return true;
     }
 
+    /**
+     * @dev set the ticker registration fee in POLY tokens
+     * @param _registrationFee registration fee in POLY tokens (base 18 decimals)
+     */
+    function setPolyRegistrationFee(uint256 _registrationFee) public onlyOwner {
```
Comment (satyamakgec) [inline contracts/TickerRegistry.sol]: change the function name as we are changing the registration fee, not setting the fee for one time
it should be named as `changePolyRegistrationFee` and also add the event for logging the same. with old value and new value as the arguments of the event.

File: contracts/ModuleRegistry.sol [hunk 3]
```diff
@@ -33,6 +34,9 @@ contract ModuleRegistry is IModuleRegistry, Ownable {
     // Emit when the module get verified by the Polymath team
     event LogModuleVerified(address indexed _moduleFactory, bool _verified);
 
+    // Payable Fallback
+    function() public payable {}
+
     /**
```
Comment (adamdossa) [inline contracts/ModuleRegistry.sol]: We shouldn't have a payable fallback function.

File: contracts/SecurityTokenRegistry.sol [hunk 4]
```diff
@@ -21,18 +26,23 @@ contract SecurityTokenRegistry is Ownable, ISecurityTokenRegistry, Util {
...
+        registrationFee = _registrationFee;
 
         // By default, the STR version is set to 0.0.1
         setProtocolVersion(_stVersionProxy, "0.0.1");
     }
 
+    // Payable Fallback
+    function() public payable {}
+
     /**
```
Comment (adamdossa) [inline contracts/SecurityTokenRegistry.sol]: We shouldn't have a payable fallback function,

File: contracts/TickerRegistry.sol [hunk 5]
```diff
@@ -43,10 +51,14 @@ contract TickerRegistry is ITickerRegistry, Ownable, Util {
...
-
+    constructor (address _polyAddress, uint256 _registrationFee) public {
+        polyAddress = _polyAddress;
+        registrationFee = _registrationFee;
     }
 
+    // Payable Fallback
+    function() public payable {}
+
     /**
```
Comment (adamdossa) [inline contracts/TickerRegistry.sol]: No payable fallback function needed.

**Review by satyamakgec — APPROVED** [review]

**Pull request merged**